frontend/appflowy_flutter/lib/mobile/presentation/database/card/row/cells/date_cell/mobile_date_cell_edit_screen.dart

```diff
@@ -0,0 +1,389 @@
+                        isEndTime: false,
+                      ),
+                    ),
+                  ],
+                )
+              : const SizedBox.shrink(),
+        );
+      },
+    );
+  }
+}
+
+class _EndDayTime extends StatelessWidget {
+  const _EndDayTime();
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<DateCellCalendarBloc, DateCellCalendarState>(
+      builder: (context, state) {
+        return AnimatedSwitcher(
+          duration: const Duration(milliseconds: 300),
+          child: state.includeTime && state.endTimeStr != null
+              ? Row(
+                  children: [
+                    Text(
+                      LocaleKeys.grid_field_endDateTime.tr(),
+                      style: Theme.of(context).textTheme.titleMedium,
+                    ),
+                    const Spacer(),
+                    // TODO(yijing): improve width
+                    SizedBox(
+                      width: 180,
+                      child: _TimeTextField(
+                        timeStr: state.timeStr,
+                        isEndTime: true,
+                      ),
+                    ),
+                  ],
+                )
+              : const SizedBox.shrink(),
+        );
+      },
+    );
+  }
+}
+
+class _TimeTextField extends StatefulWidget {
+  final bool isEndTime;
+  final String? timeStr;
+
+  const _TimeTextField({
+    required this.timeStr,
+    required this.isEndTime,
+  });
+
+  @override
+  State<_TimeTextField> createState() => _TimeTextFieldState();
+}
+
+class _TimeTextFieldState extends State<_TimeTextField> {
+  late final TextEditingController _textController;
+
+  @override
+  void initState() {
+    super.initState();
+    _textController = TextEditingController(text: widget.timeStr);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<DateCellCalendarBloc, DateCellCalendarState>(
+      listener: (context, state) {
+        if (widget.isEndTime) {
+          _textController.text = state.endTimeStr ?? "";
+        } else {
+          _textController.text = state.timeStr ?? "";
+        }
+      },
+      builder: (context, state) {
+        return TextFormField(
+          controller: _textController,
+          textAlign: TextAlign.end,
+          decoration: InputDecoration(
+            hintText: state.timeHintText,
+            errorText: widget.isEndTime
+                ? state.parseEndTimeError
+                : state.parseTimeError,
+          ),
+          keyboardType: TextInputType.datetime,
+          onFieldSubmitted: (timeStr) {
+            if (widget.isEndTime) {
+              context
+                  .read<DateCellCalendarBloc>()
+                  .add(DateCellCalendarEvent.setEndTime(timeStr));
+            } else {
+              context
+                  .read<DateCellCalendarBloc>()
+                  .add(DateCellCalendarEvent.setTime(timeStr));
+            }
```

Comment:
Personal pref, formatting is most likely way off
```suggestion
            context.read<DateCellCalendarBloc>()
              .add(
                widget.isEndTime 
                    ? DateCellCalendarEvent.setEndTime(timeStr)
                    : DateCellCalendarEvent.setTime(timeStr),
              );
```